src/ORM/DataList.php

```
@@ -308,72 +313,106 @@ public function distinct($value)
+     * @param string|array $args
      * @example $list = $list->sort('Name'); // default ASC sorting
-     * @example $list = $list->sort('Name DESC'); // DESC sorting
+     * @example $list = $list->sort('Name ASC, Age DESC');
      * @example $list = $list->sort('Name', 'ASC');
-     * @example $list = $list->sort(array('Name'=>'ASC', 'Age'=>'DESC'));
-     *
-     * @param string|array Escaped SQL statement. If passed as array, all keys and values are assumed to be escaped.
-     * @return static
+     * @example $list = $list->sort(['Name' => 'ASC', 'Age' => 'DESC']);
      */
-    public function sort()
+    public function sort(...$args): static
     {
-        $count = func_num_args();
-
+        $count = count($args);
         if ($count == 0) {
             return $this;
         }
-
         if ($count > 2) {
             throw new InvalidArgumentException('This method takes zero, one or two arguments');
         }
-
         if ($count == 2) {
-            $col = null;
-            $dir = null;
-            list($col, $dir) = func_get_args();
-
-            // Validate direction
-            if (!in_array(strtolower($dir ?? ''), ['desc', 'asc'])) {
-                user_error('Second argument to sort must be either ASC or DESC');
-            }
-
-            $sort = [$col => $dir];
+            list($column, $direction) = $args;
+            $sort = [$column => $direction];
         } else {
-            $sort = func_get_arg(0);
+            $sort = $args[0];
+            // If $sort is string then convert string to array to allow for validation
+            if (is_string($sort)) {
+                $newSort = [];
+                // Making the assumption here there are no commas in column names
+                // Other parts of silverstripe will break if there are commas in column names
+                foreach (explode(',', $sort) as $colDir) {
+                    if (empty($colDir)) {
+                        continue;
+                    }
```

This would only happen if someone has something like `MyColumn ASC,,AnotherColumn DESC`, right? So two commas sequentially? We should probably throw an exception for that case, as it is an indication that the developer has done something wrong (probably forgot to add a column in there)

---

Have removed, it's not necessary code, not sure why I had it there

> We should probably throw an exception for that case

I don't think we need to be that helpful, you'll probably get a mysql error saying that `''` is an invalid column

---

> you'll probably get a mysql error saying that `''` is an invalid column

Can you please validate that with a unit test? I think you're probably right, but the "probably" from both of us isn't ideal.

---

Have added unit test

---

Have added unit test